modules/tinymce/src/themes/silver/main/ts/backstage/Anchors.ts

```diff
@@ -15,42 +15,43 @@ const bubbleAlignments = {
   top: []
 };
 
-const getInlineDialogAnchor = (bodyElement: () => Element, lazyAnchorbar: () => AlloyComponent, useEditableAreaAnchor: boolean): () => HotspotAnchorSpec | NodeAnchorSpec => {
+const getInlineDialogAnchor = (contentAreaElement: () => Element, lazyAnchorbar: () => AlloyComponent, useEditableAreaAnchor: boolean): () => HotspotAnchorSpec | NodeAnchorSpec => {
+  const bubble = Bubble.nu(-12, 12, bubbleAlignments);
+  const overrides = {
+    maxHeightFunction: MaxHeight.expandable()
+  };
+
   const editableAreaAnchor = (): NodeAnchorSpec => ({
     anchor: 'node',
-    root: bodyElement(),
-    node: Option.from(bodyElement()),
-    bubble: Bubble.nu(-12, -12, bubbleAlignments),
+    root: Body.getBody(Traverse.owner(contentAreaElement())),
+    node: Option.from(contentAreaElement()),
+    bubble,
     layouts: {
-      onRtl: () => [ LayoutInside.northeast ],
-      onLtr: () => [ LayoutInside.northwest ]
+      onRtl: () => [ LayoutInside.northwest ],
+      onLtr: () => [ LayoutInside.northeast ]
     },
-    overrides: {
-      maxHeightFunction: MaxHeight.expandable()
-    }
+    overrides
   });
 
   const standardAnchor = (): HotspotAnchorSpec => ({
     anchor: 'hotspot',
     hotspot: lazyAnchorbar(),
-    bubble: Bubble.nu(-12, 12, bubbleAlignments),
+    bubble,
     layouts: {
       onRtl: () => [ Layout.southeast ],
       onLtr: () => [ Layout.southwest ]
     },
-    overrides: {
-      maxHeightFunction: MaxHeight.expandable()
-    }
+    overrides
   });
 
   return useEditableAreaAnchor ? editableAreaAnchor : standardAnchor;
 };
 
-const getBannerAnchor = (bodyElement: () => Element, lazyAnchorbar: () => AlloyComponent, useEditableAreaAnchor: boolean): () => HotspotAnchorSpec | NodeAnchorSpec => {
+const getBannerAnchor = (contentAreaElement: () => Element, lazyAnchorbar: () => AlloyComponent, useEditableAreaAnchor: boolean): () => HotspotAnchorSpec | NodeAnchorSpec => {
   const editableAreaAnchor = (): NodeAnchorSpec => ({
     anchor: 'node',
-    root: bodyElement(),
-    node: Option.from(bodyElement()),
+    root: Body.getBody(Traverse.owner(contentAreaElement())),
```

Comment:
There is a bit of duplication in this file. I'll defer to your judgement: does it make it easier to understand/maintain or harder?

Reply:
Previous issue unrelated to this change. I'll take a look and see if it can be improved.

Reply:
Discussed and it wouldn't make it clearer and would actually complicate this if we tried to break the small amounts of duplication out.